### ISSUE TITLE
Low(ish) framerate and visual artifacts on Rom 03 IIGS with AppleSqueezer

### ISSUE THREAD

**Issue by geoffp** [lscharen/silky-gs#5]
My setup is a ROM 03 IIGS with an AppleSqueezer v2 running core 10, booting GS/OS 6.0.4 from a CFFA3000.

Two unexpected behaviors when running accelerated:
1. There is some transient offsetting of certain horizontal rows of pixels in _moving_ sprites. Stationary sprites usually (always?) look perfect. If I disable the AppleSqueezer's acceleration, the artifacts do not occur.
2. Even when running at 14mhz, the framerate is not that different from running at 2.6 mhz. When running at 14mhz equivalent in KEGS, however, the framerate is dramatically better.

A video of running at 14mhz is attached.

https://github.com/user-attachments/assets/e7f09b21-d89d-4d20-a0ed-c66e0af9f2eb

**Comment by lscharen** [comment]
Interesting.  I sent Nick a build of Super Mario for him to test on Apple Squeezer at some point and he didn't report any issues.  I'll let him know about this and see if he'll take a look.

**Comment by lscharen** [comment]
Are you using the build from the `emu/Target.2mg` file?

**Comment by geoffp** [comment]
Yes. My POC Linux build is based on the `ice-climber` branch, since `main` wouldn't build for me.

**Comment by geoffp** [comment]
Oh -- I think I misunderstood. Do you mean there's a completed build that's present in that file when you check out the source, without actually running the build process? If so, I didn't realize that, so my answer would be no. I'm testing the result of the Linux build process.

If you'd like me to test the checked-in build, I can certainly do that.

**Comment by lscharen** [comment]
Yes, the `emu/Target.2mg` disk image in the repo has built versions of the games.

**Comment by danq989** [comment]
> My setup is a ROM 03 IIGS with an AppleSqueezer v2 running core 10, booting GS/OS 6.0.4 from a CFFA3000.
> 
> Two unexpected behaviors when running accelerated:
> 
>     1. There is some transient offsetting of certain horizontal rows of pixels in _moving_ sprites. Stationary sprites usually (always?) look perfect. If I disable the AppleSqueezer's acceleration, the artifacts do not occur.
> 
>     2. Even when running at 14mhz, the framerate is not that different from running at 2.6 mhz. When running at 14mhz equivalent in KEGS, however, the framerate is dramatically better.
 
I have a ROM 00/01 GS also with AppleSqueezer and I can confirm the exact same issues - horizontal rows offset in moving sprites and not quite as much acceleration as I would expect. Although this may be that emulators set to 14 MHz run all RAM at 14MHz and a real accelerator like AppleSqueezer has to slow down to copy data to real video DRAM so it can be displayed by the VGC.
